include/ion/node.h

```diff
@@ -87,6 +87,8 @@ class Node {
 
     void set_iport(const std::string& name, Port port);
 
+    void set_dynamic_port(Port port) ;
```

Comment:
"dynamic port" is confusing, we already have feature to add input/output dynamically ([ref](https://github.com/fixstars/ion-kit/blob/main/test/test-bb.h#L115)). Can we just say "unbound port"?